### PR TITLE
[xxxx] Add prompt for audited user in production console

### PR DESCRIPTION
**Pull request by stevehook**
### Context
We want to be able to easily attribute changes made in `rails console` on production to the developer that ran them.

### Changes proposed in this pull request
This PR adds a prompt to ask for your name when you open `rails console` in a production environment. It uses the name input to attribute all audit entries generated as a result of any subsequent commands in that console session.

image

### Guidance to review
Have I missed anything?

### Important business

- [x] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [x] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [x] Do we need to send any updates to DQT as part of the work in this PR?
- [x] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: config/initializers/console.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      end
+    end
+  end
+
+  def show_warning_message_about_environments
+    if production?
+      puts(("*" * 50).red)
+      puts("** You are in the Rails console for PRODUCTION! **".red)
+      puts(("*" * 50).red)
+    else
```
Comment (darokel) [inline config/initializers/console.rb:24]: Do we care about creating audit logs in non prod environments? I think the audit entries aren't even included in the sanitised data dump

File: config/initializers/console.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    environment_name == "production"
+  end
+
+  def development?
+    environment_name == "development"
+  end
+
+  def environment_name
+    Rails.env
+  end
```
Comment (carlosmartinez) [inline config/initializers/console.rb:41]: not too fussed, but we could probably do without these helpers and just inline them as `Rails.env.production?` and `Rails.env.development?`

**Review by carlosmartinez — APPROVED** [review]

File: config/initializers/console.rb (hunk 1 above)
Comment (stevehook) [reply config/initializers/console.rb:24]: I would say audit logs are less useful in non-production environments but not useless. 

It makes perfect sense for `audits` to be excluded from the sanitised data!

Comment (darokel) [reply config/initializers/console.rb:26]: what i meant was is there any value in enabling [this](https://github.com/DFE-Digital/register-trainee-teachers/pull/3458/files#diff-8ec81f668ca5db7966024f710de8c9cbbd40387c4b63d951ed2f1f287d8262dcR10) in anything other than prod? The other environments get wiped out daily. Almost feels like it should be inverted 🤔

**Review by defong — APPROVED** [review]

File: config/initializers/console.rb (hunk 1 above)
Comment (stevehook) [reply config/initializers/console.rb:26]: @darokel sorry! It's a fair point, the prime reason for this is better traceability for production data changes.

**Pull request merged**